### PR TITLE
GitHub sync

### PR DESCRIPTION
**Pull request by nshahan**
Big update here since there hasn't been a release for a few weeks. Should be in sync with the +12 version of angular that was released yesterday.

Updated example branch:
https://github.com/dart-lang/angular_components_example/tree/github-sync

### CODE REVIEW

File: lib/material_radio/material_radio.dart [hunk 1]
```diff
@@ -120,48 +102,38 @@ class MaterialRadioComponent extends RootFocusable
...
-  Icon _icon = uncheckedIcon;
+  Icon get icon => _checked ? checkedIcon : uncheckedIcon;
 
   /// Current tab index, depends on state of [disabled] and selection status
   /// if in group.
-  String get tabIndex => '$_tabIndex';
-  int _tabIndex = 0;
-  int _enabledTabIndex = 0;
+  @HostBinding('attr.tabindex')
+  int get tabIndex => disabled ? -1 : _enabledTabIndex;
```
Comment (chalin) [inline lib/material_radio/material_radio.dart]: In `lib/material_checkbox/material_checkbox.dart`, we have essentially:

```dart
  @HostBinding('attr.tabindex')
   String get tabIndex => disabled ? '-1' : _defaultTabIndex;
```

Should the getter return type be `String` here too?

(Should the class extend `HasTabIndex`?)

File: lib/material_radio/material_radio_group.dart [hunk 2]
```diff
@@ -60,106 +60,76 @@ import 'package:angular_components/utils/disposer/disposer.dart';
...
-    } else {
-      // Initialize tabIndex.
-      _resetTabIndex();
-    }
-  }
+  @HostBinding('attr.role')
+  static const role = 'radiogroup';
+
+  @HostBinding('attr.tabindex')
+  static const tabIndex = -1;
```
Comment (chalin) [inline lib/material_radio/material_radio_group.dart]: Should this be a string '-1'? (If so, I think that there are other instances where it is an int.)

File: lib/material_select/material_select_dropdown_item.dart [hunk 3]
```diff
@@ -55,12 +38,21 @@ import 'package:angular_components/utils/id_generator/id_generator.dart';
 )
 class MaterialSelectDropdownItemComponent extends MaterialSelectItemComponent
     implements OnDestroy {
+  @HostBinding('class')
+  static const hostClass = 'item';
+
+  // The qualified name is long because button_directive.dart uses hostTabIndex.
+  @HostBinding('tabIndex')
+  static const hostTabIndexForSelectDropdown = 0;
```
Comment (chalin) [inline lib/material_select/material_select_dropdown_item.dart]: Should this be a string?

File: lib/scorecard/scorecard.dart [hunk 4]
```diff
@@ -198,6 +196,12 @@ class ScorecardComponent extends KeyboardOnlyFocusIndicatorDirective {
     _changeDetector.markForCheck();
   }
 
+  @HostBinding('attr.tabindex')
+  int get hostTabIndex => selectable ? 0 : null;
+
+  @HostBinding('attr.role')
+  String get hostRole => selectable ? "button" : null;
```
Comment (chalin) [inline lib/scorecard/scorecard.dart]: nit: "..." -> '...'

File: lib/src/model/selection/noop_selection_model_impl.dart [hunk 5]
```diff
@@ -66,6 +66,9 @@ class _NoopSelectionModelImpl<T> implements SelectionModel<T> {
   Stream<List<SelectionChangeRecord<T>>> get selectionChanges =>
       new Stream.fromIterable(const []);
 
+  @override
+  final T selectedValue = null;
```
Comment (chalin) [inline lib/src/model/selection/noop_selection_model_impl.dart]: Why not write:

```dart
T get selectedValue => null;
```

(Come to think of it, they are probably equivalent.)

File: lib/src/model/selection/noop_selection_model_impl.dart [hunk 6]
```diff
@@ -66,6 +66,9 @@ class _NoopSelectionModelImpl<T> implements SelectionModel<T> {
...
+
   @override
   final List<T> selectedValues = const [];
```
Comment (chalin) [inline lib/src/model/selection/noop_selection_model_impl.dart]: Ditto, how about:

```dart
List<T> get selectedValues => const [];
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,60 @@
...
+ * Add `HasDisabled` interface.
+ * Propagate css classes on modal component to the overlay element.
+ * Annotate `Domservice.leap()` visible for testing.
+ * Add support for `SingleSelectionModel` to `SelectionModel.empty`.
+ * Add `mat-link-deep` mixin to push the Material link styles everywhere.
+ * Updates to new `@HostBinding` syntax.
+ * Update sass_builder dependency to latest and use the post-process builder to
+   cleanup `.scss` files from the output.
+ * Support RTL selectors on any element.
+ * Remove usage of package uuid.
```
Comment (chalin) [inline CHANGELOG.md]: nit: uuid -> UUID

File: pubspec.yaml [hunk 8]
```diff
@@ -23,6 +23,5 @@ dependencies:
   observable: '>=0.17.0+1 <0.23.0'
   protobuf: ^0.7.1
   perf_api: ^0.1.0
-  quiver: '>=0.24.0 <0.29.0'
-  sass_builder: ^1.1.2
-  uuid: ^0.5.3
+  quiver: '>=0.24.0 <0.30.0'
```
Comment (chalin) [inline pubspec.yaml]: \o/

File: pubspec.yaml [hunk 9]
```diff
@@ -23,6 +23,5 @@ dependencies:
...
+  sass_builder: ^2.0.0
```
Comment (chalin) [inline pubspec.yaml]: Do we need a dependency override for the angular compiler?

Comment (nshahan) [reply pubspec.yaml]: I haven't seen any problems. Can you elaborate on what you saw? Is it an issue with this version of the components?

I prefer to leave the version constraint up to the angular package unless something is broken. In the issue that was filed it looked like they were using mixed versions of angular and angular_components that probably were incompatible.

Comment (nshahan) [reply pubspec.yaml]: @jakemac53 showed me the issue. It seems like there was a change in angular_compiler v0.4.0-alpha+12 that breaks if you use it with angular v5.0.0-alpha+11. Releasing this version should allow to version solve and upgrade to angular v5.0.0-alpha+12 which fixes the issue. Maybe we should ask angular to pin specific versions. I don't want to have to track the transitive dependencies of angular here.

File: lib/material_radio/material_radio.dart (hunk 1 above)
Comment (nshahan) [reply lib/material_radio/material_radio.dart]: I know there is cleanup/refactor going on. I'll ask the author.

**Pull request merged**